web/source/osk/visualKeyboard.ts

```diff
@@ -1868,7 +1868,7 @@ namespace com.keyman.osk {
         this.hkKey=this.getSpecialKey(nLayer,'K_ROPT');
 
         // Always adjust screen height if iPhone or iPod, to take account of viewport changes
-        if(device.OS == 'iOS' && device.formFactor == 'phone') {
+        if(device.OS == 'iOS') {
           this.adjustHeights();
         }
```

Comment:
I'm curious why the removed condition was put in place; removing it restores the original comment's context and also is critical for resolving the issue, which tended to show up on iPad rather than iPhone.

Note that the `adjustHeights` function is also used during device rotation to resize the keyboard, and rotation had been the workaround to get the banner to display if hidden on initial load.